### DISCUSSION TITLE
(EthernetSSLClient)(SSL_ERROR)(available): Cannot operate on a closed SSL connection.

### DISCUSSION THREAD

**Discussion by simogaspa84** [khoih-prog/EthernetWebServer_SSL#27] [Q&A] [answered]
Hi @khoih-prog ...

I am getting this issue... 

Do you have an idea of the meaning and a possible way to fix it?

What should I check?

Thanks a lot

**Comment by khoih-prog** [comment, marked as answer]
HI @simogaspa84 

Please follow the guideline when posting any issue, such as [**MRE**](https://stackoverflow.com/help/minimal-reproducible-example), etc. so that anybody can duplicate it. OTW, your post will be **ignored and deleted**.

First do some **basic research**, and you'll know what it means. Be independent first then you can learn a lot from the experience. Always start from the **known working** examples or previous codes.

https://github.com/khoih-prog/EthernetWebServer_SSL/blob/fc79ef0399f898ac6fa0f3897305d52cd6f85db0/src/SSLClient/SSLClient_Impl.h#L423-L433

Also ask the general questions on general Forums or [**OPEnSLab-OSU**](https://github.com/OPEnSLab-OSU/SSLClient/issues) if SSL- related, unless you're sure that it's a bug of this library.

**Reply by simogaspa84** [reply]
Sorry @khoih-prog ..  i have tried to search for similar issues but i found only things related to the certificate.
Sorry again for bothering you 
Thanks

**Comment by khoih-prog** [comment]
Check [**Cannot operate on a closed SSL connection** #68](https://github.com/OPEnSLab-OSU/SSLClient/issues/68)

**Reply by simogaspa84** [reply]
Thanks @khoih-prog I will check because increasing the buffer size doesn't work... I was able to makee it work with one device created in aws console with one endpoint but with another one endpoint doesn't work. I will investigate on it . Thanks

**Reply by khoih-prog** [reply]
> I was able to makee it work with one device created in aws console with one endpoint but with another one endpoint doesn't work

This is **big clue**, and **only you** can find out why by spotting the differences (endpoint => SSL certs, your code, boards, etc.)

**Reply by simogaspa84** [reply]
i know @khoih-prog and this is why I started bothering you for the trusted anchor arrays.. 
I tried to create by myself by calling the python script and it is the same in both the boards.. 
The two endpoints are different (the working one in the us and the wrong one in europe) ...  thingname are different too but 
i don't see big issues here.. 
Moreover the test with wifi instead ethernet is working also with the new device...
The device certificate and the private key are generated from aws and I just copy and paste.. 
I really don't know it seems to be sometihng related about aws.. 
Thank thanks a lot again for your time 
Sorry again @khoih-prog

**Reply by khoih-prog** [reply]
> The two endpoints are different (the working one in the us and the wrong one in europe) 

Your better and remaining choice is to ask AWS people why. I'm afraid nobody else can help you.

**Reply by simogaspa84** [reply]
yes @khoih-prog  I will contact aws and then I will let you know if you want ... Thanks

**Reply by khoih-prog** [reply]
Sure we'd appreciate if you inform us the real issue and solution, and learn something from the experience.

**Reply by simogaspa84** [reply]
But now we can share with the community that if they will see closed ssl connection to check if they have some filters in the topics inserted in the aws console.

**Comment by khoih-prog** [comment]
Try to increase `m_iobuf` from

https://github.com/khoih-prog/EthernetWebServer_SSL/blob/fc79ef0399f898ac6fa0f3897305d52cd6f85db0/src/SSLClient/SSLClient.h#L519-L521

to 4096, 8192 or larger to see if OK

```
    // KH test increase
    //unsigned char m_iobuf[2048];
    unsigned char m_iobuf[4096];
```

If not, you have to put more debug and know what's wrong.

**Comment by khoih-prog** [comment]
Hi @simogaspa84 

Have a look at `Certificate bundles for AWS Regions` section in [**Using SSL/TLS**](https://docs.aws.amazon.com/AmazonRDS/latest/UserGuide/UsingWithRDS.SSL.html), and try to use the Certs for the interested European Region, and see if OK

Got news from AWS yet?

**Reply by simogaspa84** [reply]
Hi @khoih-prog thanks again for your support... Yes I had the contact from aws that told me to check if there was simply some filters in the topic coming from the iot board. 
I did personally the account and the setting for the device in the us region. 
The device in the europe region was created and managed from a consultant that basically forgot to remove some filters in the topic, 
Basically it was him rejecting the connection because he didn't like the name of the topic.. 
It was very trivial and the end.
But now we can share with the community that if they will see closed ssl connection to check if they have some filters in the topics inserted in the aws console.
Sorry sorry again for bothering you. I really appreciate your help and I hope to work again with you because you have a smart mind @khoih-prog
